task8/TutterApp/src/main/java/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,93 @@
+package dao;
+
+import models.User;
+
+import javax.naming.NamingException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+
+public class UserDAOImpl implements UserDAO {
+    private static Logger logger;
+    private static ConnectionPool connectionPool;
+
+    static {
+        try {
+            LogManager.getLogManager().readConfiguration(UserDAOImpl.class.getClassLoader().getResourceAsStream("logging.properties"));
+
+            logger = Logger.getLogger(UserDAOImpl.class.getName());
+        } catch (Exception ignored) {
```

Comment:
!
+ as it goes by.... the quality of code directly depends on how may timed you copy-paste same structre, o statement, or block. Ideally, nothing at all copypasted. It still happens sometime ... But every time before making a PR look back to your code and ask yourself, haven't I missed a chance to abstract something, to create a common superclass or an utility class.. etc.

> "Perfection is not when you have nothing to add. Perfection is when you have nothing to delete!")